apps/api/src/app/workflows-v2/usecases/build-step-issues/build-step-issues.usecase.ts

```diff
@@ -86,8 +88,15 @@ export class BuildStepIssuesUsecase {
     const skipLogicIssues = sanitizedControlValues?.skip
       ? this.validateSkipField(sanitizedControlValues.skip as RulesLogic<AdditionalOperation>)
       : {};
-
-    return merge(schemaIssues, liquidIssues, customIssues, skipLogicIssues);
+    const integrationIssues = await this.validateIntegration({
+      stepTypeDto,
+      environmentId: user.environmentId,
+      organizationId: user.organizationId,
+      workflowId: persistedWorkflow!._id,
+      stepInternalId: stepInternalId!,
```

Comment:
@djabarovgeorge How come these are nullish in this context?

Reply:
This is a valid question, int the current context. 

`stepInternalId` is optional because we handle issues via an upsert operation meaning a single create/update action. During the creation phase, `stepInternalId` does not exist. It is primarily used for two purposes:

1. **`buildAvailableVariableSchemaUsecase`**: To generate the variable schema for previous steps.
2. **Fetching Persistent Control Values**: To retrieve persisted control values if they are not provided dynamically.

do you have any ideas on how to simplify this flow to make it more strict?